api/host.go

```diff
@@ -138,3 +143,91 @@ func GetUpdateHostStatusCallback(rq amboy.Queue, status string, notes string, us
 		return httpStatus, modifyErr
 	}
 }
+
+// TerminateSpawnHost is a shared utility function to terminate a spawn host
+func TerminateSpawnHost(ctx context.Context, env evergreen.Environment, h *host.Host, u *user.DBUser, r *http.Request) (*host.Host, int, error) {
+	if h.Status == evergreen.HostTerminated {
+		err := errors.New(fmt.Sprintf("Host %v is already terminated", h.Id))
+		return nil, http.StatusInternalServerError, err
```

Comment:
This was also wrong before, but this should really be bad request. Same with the other checks that the host's status isn't already in the target status